Add version 3 release section to changelog (#798)
appsignal/appsignal-nodejs: CHANGELOG.md

# AppSignal for Node.js Changelog

## 3.0.0

Welcome to AppSignal for Node.js version 3.0.0.!

> In this release, we have switched to using OpenTelemetry internally for tracing. (We do not yet support OpenTelemetry for metrics and logging.) It is no longer needed to use AppSignal instrumentation packages like `@appsignal/express`, `@appsignal/koa`, `@appsignal/nextjs`, `@appsignal/apollo-server`, etc. Some official OpenTelemetry instrumentation packages are bundled with our package as dependencies and instrumented automatically. Other _supported instrumentation packages_ can be [added and configured manually](https://docs.appsignal.com/nodejs/3.x/instrumentation/additional-instrumentations.html).

Reviewer: I think it's good to say that we still provide support for metrics in this version.

We strongly recommend following the [migration guide] while upgrading. Instrumentation from package version 2 is not compatible with version 3. Custom instrumentation will need to be set up using [OpenTelemetry instrumentation](https://docs.appsignal.com/nodejs/3.x/instrumentation/instrumentation.html).

If you run into any issues upgrading to this new version, please contact us at [support@appsignal.com][support] for help.

- [Version 3 migration guide][migration guide]
- [Contact us for support][support]

[migration guide]: https://docs.appsignal.com/nodejs/3.x/migration-guide.html
[support]: mailto:support@appsignal.com

> ### Added

Reviewer: We also support [Next.js](https://docs.appsignal.com/nodejs/3.x/integrations/nextjs.html)

# AppSignal for Node.js Changelog

## 3.0.0

Welcome to AppSignal for Node.js version 3.0.0.!

> In this release, we have switched to using OpenTelemetry internally for tracing. (We do not yet support OpenTelemetry for metrics and logging, but do continue to support metrics through our own metrics system.) It is no longer needed to use AppSignal instrumentation packages like `@appsignal/express`, `@appsignal/koa`, `@appsignal/nextjs`, `@appsignal/apollo-server`, etc. Some official OpenTelemetry instrumentation packages are bundled with our package as dependencies and instrumented automatically. Other _supported instrumentation packages_ can be [added and configured manually](https://docs.appsignal.com/nodejs/3.x/instrumentation/additional-instrumentations.html).

Reviewer 2: ```suggestion
In this release, we have switched to using OpenTelemetry internally for tracing. (We do not yet support OpenTelemetry for metrics and logging, but do continue to support metrics through our own metrics system.) The AppSignal instrumentation packages `@appsignal/express`, `@appsignal/koa`, `@appsignal/nextjs` and `@appsignal/apollo-server` should no longer be used. Some official OpenTelemetry instrumentation packages are bundled with our package as dependencies and instrumented automatically. Other _supported instrumentation packages_ can be [added and configured manually](https://docs.appsignal.com/nodejs/3.x/instrumentation/additional-instrumentations.html).
```

Less of a "not needed" and more of a "please don't". Also, that's the whole list of packages.

We strongly recommend following the [migration guide] while upgrading. Instrumentation from package version 2 is not compatible with version 3. Custom instrumentation will need to be set up using [OpenTelemetry instrumentation](https://docs.appsignal.com/nodejs/3.x/instrumentation/instrumentation.html).

If you run into any issues upgrading to this new version, please contact us at [support@appsignal.com][support] for help.

- [Version 3 migration guide][migration guide]
- [Contact us for support][support]

[migration guide]: https://docs.appsignal.com/nodejs/3.x/migration-guide.html
[support]: mailto:support@appsignal.com

### Added

- Add support for express package
- Add support for graphql package
- Add support for ioredis package
- Add support for Fastify
- Add support for koa package
- Add support for mysql package
- Add support for mysql2 package
- Add support for MongoDB
- Add support for Mongoose
- Add support for Nest.js
- Add support for Next.js
- Add support for pg package
- Add support for prisma package
- Add support for redis package
- Add support for redis package version 4
- Add support for the fs module

> - Instrument outgoing HTTP requests

Reviewer 2: Do we want to mention here packages that were already supported on 2.4? If not, drop Express, Koa, Postgres, Redis and outgoing HTTP requests.

If we also want to exclude those who were supported as part of the 2.4 OpenTelemetry beta, then also drop Redis version 4, IORedis, MySQL and MySQL2.

Author: I don't know. We're also mentioning packages we don't explicitly support, but are supported through the HTTP instrumentation package 🤷‍♂️

Reviewer 2: I'm not sure which ones you mean?

Author: I can't find any special instrumentation or extractors for Next.js, for example. We support it, but only through other instrumentation.

I thought we talked about providing a list of things we support in this changelog update last Monday. We needed a list of what we do support. I don't know to what extend we want to include any HTTP library/framework that we indirectly support.

Reviewer 2: Ah, right. Next.js is a special case there 😅 I wouldn't mention it because we already supported it.

Anyway, just writing a comment to remind you to add Knex to it.
